### PR TITLE
feat: Vulcan Nofall and Better Spider

### PR DESCRIPTION
**Pull request by Razzy52**

### CODE REVIEW

**Comment by EclipsesDev** [comment]
do u mean `2.8.8` ? 👀

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/spider/modes/SpiderVulcan286.kt [hunk 1]
```diff
@@ -22,6 +22,22 @@ import net.ccbluex.liquidbounce.config.Choice
...
+import net.ccbluex.liquidbounce.utils.math.toBlockPos
+import net.minecraft.block.Blocks
+import net.minecraft.network.packet.c2s.play.ClientCommandC2SPacket
+import net.minecraft.network.packet.c2s.play.PlayerMoveC2SPacket
+import net.minecraft.util.math.BlockPos
+import net.minecraft.util.shape.VoxelShapes
+import kotlin.math.cos
+import kotlin.math.sin
 
 internal object SpiderVulcan286 : Choice("Vulcan") {
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/spider/modes/SpiderVulcan286.kt]: put the version into the name

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/spider/modes/SpiderVulcan288.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        if (stop && !player.horizontalCollision) {
+            player.velocity.y = 0.0
+            stop = false
+
+        }
+        // if (collidesVertical()) {
+            //player.velocity.y = 0.0
+        //}
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/spider/modes/SpiderVulcan288.kt]: Remove when not used.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/spider/modes/SpiderVulcan288.kt [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            packet.z += cos(yaw) * 0.000000001
+            glitch = false
+            packet.onGround = true
+
+        }
+    }
+    //private fun collidesVertical() =
+       // world.getBlockCollisions(player, player.boundingBox.offset(0.0, 0.5, 0.0)).any { shape ->
+         //   shape != VoxelShapes.empty()
+        //}
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/movement/spider/modes/SpiderVulcan288.kt]: Remove when not used.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/nofall/modes/NoFallVulcanTP.kt [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                repeat(ticksToVoid) {
+                    simulatedPlayer.tick()
+                }
+
+                return simulatedPlayer.pos.y < voidThreshold
+            }
+        }
+
+        return false
+    }
```
Comment (1zun4) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/nofall/modes/NoFallVulcanTP.kt]: way too overkill for void detection

Comment (EclipsesDev) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/nofall/modes/NoFallVulcanTP.kt]: I agree, I ported this to legacy and I just use `fallingPlayer.findCollision(500) != null` for void checking

**Comment by Razzy52** [comment]
will work on this in a few days, have some more bypasses aswell

**Pull request merged**